osu.Game/Screens/Ranking/SoloResultsScreen.cs

```diff
@@ -1,90 +1,130 @@
+
+            // if we haven't encountered a match for the presented score, we still need to attach it.
+            // note that the above block ensuring that the `Score` reference makes it in here makes this valid to write in this way.
+            if (!sortedScores.Contains(Score))
+                sortedScores.Add(Score);
 
-                    if (score.MatchesOnlineID(Score))
+            sortedScores = sortedScores.OrderByTotalScore().ToList();
+
+            int delta = 0;
+            bool isPartialLeaderboard = leaderboardManager.CurrentCriteria?.Scope != BeatmapLeaderboardScope.Local && result.TopScores.Count >= 50;
+
+            for (int i = 0; i < sortedScores.Count; i++)
+            {
+                var sortedScore = sortedScores[i];
+
+                if (!isPartialLeaderboard)
+                    sortedScore.Position = i + 1;
+                else
+                {
+                    if (ReferenceEquals(sortedScore, Score) && sortedScore.Position == null)
                     {
-                        // we don't want to add the same score twice, but also setting any properties of `Score` this late will have no visible effect,
-                        // so we have to fish out the actual drawable panel and set the position to it directly.
-                        var panel = ScorePanelList.GetPanelForScore(Score);
-                        Score.Position = panel.ScorePosition.Value = position;
+                        int? previousScorePosition = i > 0 ? sortedScores[i - 1].Position : 0;
+                        int? nextScorePosition = i < result.TopScores.Count - 1 ? sortedScores[i + 1].Position : null;
+
+                        if (previousScorePosition != null && nextScorePosition != null && previousScorePosition + 1 == nextScorePosition)
+                        {
+                            sortedScore.Position = previousScorePosition + 1;
+                            delta += 1;
+                        }
+                        else
+                            sortedScore.Position = null;
                     }
                     else
-                    {
-                        var converted = score.ToScoreInfo(rulesets, Beatmap.Value.BeatmapInfo);
-                        converted.Position = position;
-                        toDisplay.Add(converted);
-                    }
+                        sortedScore.Position += delta;
                 }
-
-                return toDisplay.ToArray();
             }
-            catch (Exception ex)
-            {
-                Logger.Log($"Failed to fetch scores (beatmap: {Score.BeatmapInfo}, ruleset: {Score.Ruleset}): {ex}");
-                return [];
-            }
-        }
 
-        protected override void Dispose(bool isDisposing)
-        {
-            base.Dispose(isDisposing);
+            // there's a non-zero chance that the `Score`'s `ScorePosition` was mutated above,
+            // but the two are not actually coupled together in any way,
```

Comment:
Does this part of the comment still apply? Not clear where/how `ScorePosition` would be updated here.

Reply:
It does but I bungled the wording. See if https://github.com/ppy/osu/pull/32980/commits/be913927602f0e96f3e4349aa2c119fb6e7c89ec helps.